src/main/app/src/pages/BrowsePage.js

```diff
@@ -5,16 +5,47 @@ import GameCard from '../constants/GameCard.js';
 import ArticleCard from '../constants/ArticleCard.js';
 import VideoCard from '../constants/VideoCard.js';
 import ActivitiesFilterBar from '../constants/ActivitiesFilterBar';
+import LoadingIndicator from '../constants/LoadingIndicator';
+import TablePagination from '@material-ui/core/TablePagination';
 
 /* Component for browse page */
 const BrowsePage = ({ links, activityType, updateActivityType, updateActivity, updateServlet,
   articleData, videoData, gameData, activityTypes }) => {
 
   const [filteredLinks, updateFilteredLinks] = React.useState(links);       // The links or filtered links after user clicks filter button.
+  const [loading, updateLoading] = React.useState(false); // Controls display of loading indicator.
+
+  const [pageNumber, updatePageNumber] = React.useState(0); // Page number of pagination
+  const [activitiesPerPage, updateActivitiesPerPage] = React.useState(10);
+
+  // Gets the activities to display for the current page number
+  const getPageLinks = () => {
+    const links = [];
+    let countLinks = activitiesPerPage;
+    for (let idx=pageNumber*activitiesPerPage; idx < filteredLinks.length && countLinks >= 0; idx++) {
+      links.push(filteredLinks[idx]);
+      countLinks--;
+    }
+    return links;
+  }
+
+  const handleActivitiesPerPageChange = evt => {
+    updateActivitiesPerPage(parseInt(evt.target.value, 10));
+    updatePageNumber(0);
+    udpatePageFilteredLinks(getPageLinks());
+  }
+
+  const [pageFilteredLinks, udpatePageFilteredLinks] = React.useState(getPageLinks()); // links for current page only.
```

Comment:
It would be better to add all the states for a particular component right at the top where you declare other states.

Reply:
I think this is Steve's code so I'll let him change it on his commit! :)

Reply:
That wasn't possible to do in this case because I needed to call a function to provide the default value for the state. The function needs to be initialized before it's called so I had to declare it before the state. I can try to move all these helper functions out of the component to improve readability.